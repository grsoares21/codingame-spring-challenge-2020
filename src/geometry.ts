export interface Point {
  x: number;
  y: number;
}

export function manhattanDistance(pointA: Point, pointB: Point): number {
  return Math.abs(pointA.x - pointB.x) + Math.abs(pointA.y - pointB.y);
}
