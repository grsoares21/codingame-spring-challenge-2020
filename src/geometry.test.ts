import { findPath } from "./geometry";

it(`should get shortest past with A*`, () => {
  expect(
    findPath(getMap(), [], { x: 0, y: 0 }, { x: 4, y: 3 })
  ).toMatchSnapshot();
});

it(`should get shortest past with A* with pac in the way`, () => {
  expect(
    findPath(getMap(), [{ x: 3, y: 1 }], { x: 0, y: 0 }, { x: 4, y: 3 })
  ).toMatchSnapshot();
});

function getMap(): string[][] {
  return [
    [" ", " ", " ", " ", " "],
    [" ", "#", "#", " ", "#"],
    [" ", "#", "#", " ", "#"],
    [" ", " ", "#", " ", " "],
    ["#", " ", " ", " ", " "],
  ];
}
