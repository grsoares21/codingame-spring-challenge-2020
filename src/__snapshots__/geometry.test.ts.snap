// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get shortest past with A* 1`] = `
Array [
  Object {
    "x": 0,
    "y": 0,
  },
  Object {
    "x": 1,
    "y": 0,
  },
  Object {
    "x": 2,
    "y": 0,
  },
  Object {
    "x": 3,
    "y": 0,
  },
  Object {
    "x": 3,
    "y": 1,
  },
  Object {
    "x": 3,
    "y": 2,
  },
  Object {
    "x": 3,
    "y": 3,
  },
  Object {
    "x": 4,
    "y": 3,
  },
]
`;

exports[`should get shortest past with A* with pac in the way 1`] = `
Array [
  Object {
    "x": 0,
    "y": 0,
  },
  Object {
    "x": 0,
    "y": 1,
  },
  Object {
    "x": 0,
    "y": 2,
  },
  Object {
    "x": 0,
    "y": 3,
  },
  Object {
    "x": 1,
    "y": 3,
  },
  Object {
    "x": 1,
    "y": 4,
  },
  Object {
    "x": 2,
    "y": 4,
  },
  Object {
    "x": 3,
    "y": 4,
  },
  Object {
    "x": 3,
    "y": 3,
  },
  Object {
    "x": 4,
    "y": 3,
  },
]
`;
